### ISSUE TITLE
Issue adding support for Bridgeless, Lottie React Native

### ISSUE THREAD

**Issue by TheRogue76** [facebook/react-native#42477]
### Description

Hi there.

I have been working on adding bridgeless support for Lottie React Native, and i have hit a rather strange issue.
When i export my fabric component using 
```ts
export default codegenNativeComponent<NativeProps>('LottieAnimationView') as HostComponent<NativeProps>
```
The app succeeds in running in bridgeless mode, and shows the UI and the Fabric events are sent back correctly, but i get a console.error: `Native Component 'LottieAnimationView' that calls codegenNativeComponent was not code generated at build time. Please check its definition` (I have traced this console error [to here](https://github.com/facebook/react-native/blob/a58ec074b65149fdcd07383494d9273a0769111a/packages/react-native/Libraries/Utilities/codegenNativeComponent.js#L38) in React Native internals)
But it runs correctly and both shows the Fabric component and sends back the correct events
```bash
 (NOBRIDGE) ERROR  Native Component 'LottieAnimationView' that calls codegenNativeComponent was not code generated at build time. Please check its definition.
 (NOBRIDGE) LOG  Running "example" with {"rootTag":1,"initialProps":null,"fabric":true}
 (NOBRIDGE) LOG  Finished
```
Looking at the generated LottieAnimationView, the output seems correct to me as far as i can tell.
if i change the signature to
```ts
export default codegenNativeComponent<NativeProps>('LottieAnimationView')
```
The bridgeless console error goes away, but it throws an error when the event needs to come back:
```bash
 (NOBRIDGE) ERROR  Error: Unsupported top level event type "topAnimationFinish" dispatched
```
What am i missing here? The only way i have found to register the events without this error has been to put them in `customDirectEventTypes` from `react-native/Libraries/Renderer/shims/ReactNativeViewConfigRegistry`, but that looks like a very internal API, and from what i understood it registers it at the global scope, which is not ideal because someone else might also have the same event name or some other thing might happen

### Steps to reproduce

1. Clone the repo
2. run `yarn` in the root
3. cd apps/fabric
5. run `bundle install && RCT_NEW_ARCH_ENABLED=1 pod update --project-directory=ios`
6. run `yarn start`
7. build the app, either iOS or Android (Feel free to use Xcode or the commands in `package.json`)
8. Observe the error

### React Native Version

0.73.1

### Affected Platforms

Runtime - Android, Runtime - iOS

### Areas

Bridgeless - The New Initialization Flow

### Output of `npx react-native info`

```text
System:
  OS: macOS 14.2.1
  CPU: (10) arm64 Apple M1 Max
  Memory: 66.53 MB / 32.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.10.0
    path: ~/.nvm/versions/node/v20.10.0/bin/node
  Yarn:
    version: 3.2.3
    path: ~/.yarn/bin/yarn
  npm:
    version: 5.1.0
    path: ~/Desktop/lottie-react-native/node_modules/.bin/npm
  Watchman: Not Found
Managers:
  CocoaPods:
    version: 1.14.0
    path: /Users/Parsa/.rvm/gems/ruby-3.1.2/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.2
      - iOS 17.2
      - macOS 14.2
      - tvOS 17.2
      - visionOS 1.0
      - watchOS 10.2
  Android SDK:
    API Levels:
      - "26"
      - "28"
      - "29"
      - "30"
      - "31"
      - "32"
      - "33"
      - "34"
    Build Tools:
      - 28.0.3
      - 29.0.2
      - 30.0.2
      - 30.0.3
      - 31.0.0
      - 32.0.0
      - 32.1.0
      - 33.0.0
      - 33.0.1
      - 34.0.0
    System Images:
      - android-33 | Google APIs ARM 64 v8a
      - android-33 | Google Play ARM 64 v8a
      - android-34 | Google APIs ARM 64 v8a
      - android-34 | Google Play ARM 64 v8a
    Android NDK: Not Found
IDEs:
  Android Studio: Not Found
  Xcode:
    version: 15.2/15C500b
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.7
    path: /usr/bin/javac
  Ruby:
    version: 3.1.2
    path: /Users/Parsa/.rvm/rubies/ruby-3.1.2/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.73.1
    wanted: 0.73.1
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: true
  newArchEnabled: true
```

### Stacktrace or Logs

```text
(NOBRIDGE) ERROR  Native Component 'LottieAnimationView' that calls codegenNativeComponent was not code generated at build time. Please check its definition.
 (NOBRIDGE) LOG  Running "example" with {"rootTag":1,"initialProps":null,"fabric":true}
 (NOBRIDGE) LOG  Finished
```

### Reproducer

https://github.com/lottie-react-native/lottie-react-native/tree/feature/add-experimental-bridgeless

### Screenshots and Videos

Running in Bridgeless mode, with the default export cast `as HostComponent<NativeProps>`
<img width="1728" alt="Screenshot 2024-01-21 at 11 26 03" src="https://github.com/facebook/react-native/assets/40071952/a6f69444-010b-4cbb-9e10-1ef4ac89ce69">
<img width="1728" alt="Screenshot 2024-01-21 at 12 01 58" src="https://github.com/facebook/react-native/assets/40071952/3ea709b8-7511-4113-90c6-549318ae1b5f">
Running in Bridgeless mode, WITHOUT the default export cast `as HostComponent<NativeProps>`
<img width="1728" alt="Screenshot 2024-01-21 at 11 36 32" src="https://github.com/facebook/react-native/assets/40071952/2a0b2cfb-2d75-49c8-b073-fd4961c01546">

**Comment by TheRogue76** [comment]
> ⚠️	Missing Reproducible Example
> ℹ️	We could not detect a reproducible example in your issue report. Please provide either:
> * If your bug is UI related: a [Snack](https://snack.expo.dev)
> * If your bug is build/update related: use our [Reproducer Template](https://github.com/react-native-community/reproducer-react-native/generate). A reproducer needs to be in a GitHub repository under your username.

The reproducer is not under my name, but the package itself. Ignoring this error.

**Comment by TheRogue76** [comment]
The pull request that is currently in the works is https://github.com/lottie-react-native/lottie-react-native/pull/1160/files so if you do not wish to read the whole thing, just the changes, you can read them there

**Comment by TheRogue76** [comment]
@cortinico @cipolleschi (Sorry for pinging you guys directly) any thoughts and help on this is appreciated. I guess the Issue tracker is not pinging since it is still saying there is no reproducible.

**Comment by cortinico** [comment]
We discussed this internally briefly. We'll get back to you in the next future to see what we can do to support 👍

**Comment by TheRogue76** [comment]
Sounds good. Putting this on hold for now

**Comment by dmytrorykun** [comment]
```
(NOBRIDGE) ERROR  Native Component 'LottieAnimationView' that calls codegenNativeComponent was not code generated at build time. Please check its definition.
```
👆This issue has been already fixed by adding support for the `as` expression for TypeScript: https://github.com/facebook/react-native/blob/main/packages/babel-plugin-codegen/index.js#L87-L95
The fix should be available with react-native 0.74. Alternatively you can try the latest nightly version of `@react-native/babel-plugin-codegen`, the fix is there.

```
 (NOBRIDGE) ERROR  Error: Unsupported top level event type "topAnimationFinish" dispatched
```
👆And I'm currently investigating this.

**Comment by TheRogue76** [comment]
Thank you @dmytrorykun. I guess that means there is no plans to backport this change for 0.73 and below?

**Comment by dmytrorykun** [comment]
@TheRogue76 I guess we should do this, yeah.
https://github.com/reactwg/react-native-releases/discussions/98#discussioncomment-8222951

**Comment by dmytrorykun** [comment]
@TheRogue76 the quick fix would be to delete the 'onAnimationFinish', 'onAnimationFailure', and 'onAnimationLoaded' arguments from [here](https://github.com/lottie-react-native/lottie-react-native/blob/feature/add-experimental-bridgeless/packages/core/src/specs/LottieAnimationViewNativeComponent.ts#L59). I'm currently investigating the underlying problem.

**Comment by TheRogue76** [comment]
@dmytrorykun That's intriguing. I'll test it out locally. Please keep me posted on the investigation.

**Comment by TheRogue76** [comment]
Hi @dmytrorykun, is it ok if i update the PR? Now that 73.3 is out i want to move things forward, but i would totally understand if you need more time with this. I can redo this in another PR so that you would have the branch to investigate freely as well

**Comment by dmytrorykun** [comment]
Hi @TheRogue76 , sorry, I forgot to post the update here. This other issue has also been fixed, and pick is requested for 0.73.4 (https://github.com/reactwg/react-native-releases/discussions/101#discussioncomment-8290856).

**Comment by TheRogue76** [comment]
> Hi @TheRogue76 , sorry, I forgot to post the update here. This other issue has also been fixed, and pick is requested for 0.73.4 (https://github.com/reactwg/react-native-releases/discussions/101#discussioncomment-8290856).

Perfect! Thank you! Marking the issue as resolved, since every variant has been addressed.
